src/main/java/org/osc/controller/nsc/api/NeutronSdnRedirectionApi.java

```diff
@@ -31,58 +36,91 @@
 import org.osc.sdk.controller.element.NetworkElement;
 import org.osc.sdk.controller.element.VirtualizationConnectorElement;
 import org.osc.sdk.controller.exception.NetworkPortNotFoundException;
+import org.osgi.service.transaction.control.TransactionControl;
 
 public class NeutronSdnRedirectionApi implements SdnRedirectionApi {
 
-    Logger log = Logger.getLogger(NeutronSdnRedirectionApi.class);
+    private static final Logger LOG = Logger.getLogger(NeutronSdnRedirectionApi.class);
 
     private VirtualizationConnectorElement vc;
     private String region;
 
+    private TransactionControl txControl;
+    private EntityManager em;
+    private NSCUtils utils;
+
     public NeutronSdnRedirectionApi() {
     }
 
-    public NeutronSdnRedirectionApi(VirtualizationConnectorElement vc, String region) {
+    public NeutronSdnRedirectionApi(VirtualizationConnectorElement vc, String region, TransactionControl txControl,
+            EntityManager em) {
         this.vc = vc;
         this.region = region;
+        this.txControl = txControl;
+        this.em = em;
+        this.utils = new NSCUtils(em, txControl);
```

Comment:
you are using NSC utils statically as well as using an instance. `NSCUtils.makeInspectionHookElement` we need to pick one. using it as a instance class makes sense